### ISSUE TITLE
Phase 50.3: MetricMetaLearner — Prototypical, Matching & Relation Networks

### ISSUE THREAD

**Issue by web3guru888** [web3guru888/asi-build#972]
### Summary
Implement the `MetricMetaLearner` module providing metric-based few-shot learning methods: Prototypical Networks, Matching Networks, and Relation Networks. These approaches learn embedding spaces where classification reduces to nearest-neighbor or learned-distance computation.

### Motivation
Metric-based meta-learning avoids inner-loop optimization entirely, instead learning an embedding function φ such that distances in embedding space correspond to semantic similarity. At test time, few-shot classification is performed by computing distances between query embeddings and class prototypes (or support examples), enabling extremely fast inference without gradient updates.

### Architecture

```
┌────────────────────────────────────────────────────────┐
│                  MetricMetaLearner                     │
│                                                        │
│  ┌──────────────────────────────────────────────────┐  │
│  │              Embedding Network φ                  │  │
│  │  CNN / ResNet-12 / WRN-28-10 backbone            │  │
│  └─────────────────────┬────────────────────────────┘  │
│                        │ embeddings                    │
│  ┌─────────┬───────────┼───────────┬─────────────┐     │
│  │         │           │           │             │     │
│  │  ┌──────▼─────┐ ┌───▼──────┐ ┌─▼───────────┐ │     │
│  │  │ Prototypical│ │ Matching │ │  Relation   │ │     │
│  │  │ Network    │ │ Network  │ │  Network    │ │     │
│  │  │            │ │          │ │             │ │     │
│  │  │ c_k = mean │ │ attention│ │ r(φ(x),    │ │     │
│  │  │   φ(x_i)  │ │ cosine   │ │   φ(x'))   │ │     │
│  │  │ d(φ(q),c) │ │ softmax  │ │ learned    │ │     │
│  │  └────────────┘ └──────────┘ └─────────────┘ │     │
│  └───────────────────────────────────────────────┘     │
│                                                        │
│  ┌──────────────────────────────────────────────────┐  │
│  │           Distance / Similarity Module            │  │
│  │  Euclidean │ Cosine │ Mahalanobis │ Learned      │  │
│  └──────────────────────────────────────────────────┘  │
└────────────────────────────────────────────────────────┘
```

### Core API

```python
class MetricMetaLearner:
    def __init__(self, backbone: nn.Module, config: MetricMetaConfig):
        """Initialize with embedding backbone and metric configuration."""
    
    def compute_prototypes(self, support_x: Tensor, support_y: Tensor) -> Tensor:
        """Compute class prototypes as mean embeddings: c_k = (1/K) Σ φ(x_i)."""
    
    def classify(self, query_x: Tensor, support_x: Tensor,
                support_y: Tensor) -> ClassificationResult:
        """Classify query examples using the configured metric method."""
    
    def train_step(self, episodes: List[Episode]) -> TrainResult:
        """Episodic training step optimizing embedding space."""
    
    def set_method(self, method: str = "prototypical") -> None:
        """Switch: 'prototypical', 'matching', 'relation', 'proto_mahalanobis'."""

@dataclass
class MetricMetaConfig:
    method: str = "prototypical"       # 'prototypical', 'matching', 'relation'
    distance: str = "euclidean"        # 'euclidean', 'cosine', 'mahalanobis'
    embedding_dim: int = 512
    temperature: float = 1.0           # softmax temperature scaling
    backbone: str = "resnet12"         # 'conv4', 'resnet12', 'wrn2810'
    use_task_specific_adapter: bool = False

@dataclass
class ClassificationResult:
    logits: Tensor                     # [N_query, N_way]
    predictions: Tensor                # [N_query]
    probabilities: Tensor              # [N_query, N_way]
    accuracy: float
    confidence: Tensor                 # [N_query] max probability
```

### Key Features
- **Prototypical Networks** — class prototype computation + Euclidean/cosine distance
- **Matching Networks** — full context embeddings with attention kernel
- **Relation Networks** — learned pairwise relation scoring via neural network
- **Multiple distance metrics** — Euclidean, cosine, Mahalanobis, learned
- **Temperature scaling** for calibrated confidence estimates
- **Task-conditioned adaptation** — optional FiLM layers for task-specific modulation
- **Transductive inference** — leverage unlabeled query statistics (optional)
- **Backbone zoo** — Conv4, ResNet-12, WRN-28-10 with pretrained options

### Acceptance Criteria
- [ ] 5-way 5-shot miniImageNet accuracy ≥ 68% (Prototypical Networks)
- [ ] 5-way 1-shot miniImageNet accuracy ≥ 49% (Prototypical Networks)
- [ ] Relation Network matches Prototypical within 2% on standard benchmarks
- [ ] Inference speed: ≤ 5ms per episode on GPU (no inner-loop needed)
- [ ] Support both inductive and transductive inference modes
- [ ] Embedding visualization (t-SNE/UMAP) confirming cluster separation
- [ ] Unit tests with ≥ 95% coverage, integration with Phase 50.1

### Academic References
- Snell, Swersky, & Zemel (2017) — Prototypical Networks
- Vinyals et al. (2016) — Matching Networks for One Shot Learning
- Sung et al. (2018) — Relation Network for Few-Shot Learning
- Oreshkin, López, & Lacoste (2018) — "TADAM: Task Dependent Adaptive Metric"
- Bateni et al. (2020) — "Improved Few-Shot Visual Classification" (distribution calibration)
- Liu et al. (2020) — "Negative Margin Matters" (embedding optimization)

### Dependencies
- Phase 50.1 (TaskDistributionSampler) for episode generation
- Phase 46.1 (ContrastiveLearner) for pretrained embedding initialization
